### PR TITLE
Apply various user experience improvements

### PR DESCRIPTION
**Pull request by sdeken**
This does two things to improve usability:

* **Show/hide the dock icon without needing to relaunch the app** - right now, we're modifying the installed app's Info.plist file directly to turn LSUIElement on or off.  In the bad old days, this was... ok*-ish*... because although it was sort of inconvenient, at least it worked.  Somewhere along the line, though, Apple changed the strictness of their requirements for the Accessibility access, and made it so if the app bundle changed, the access was no longer valid.  As a result, you couldn't just show or hide the dock icon any more - you have to show or hide the dock icon, relaunch the app, open up the control panel, uncheck the box for KeyCastr, and check the box next to KeyCastr again.  Ugh.  
  
This used to be the only way to do it, but now there's a better way - we can just use TransformProcessType to show or hide the icon.  There are quirks - the preference window loses focus, for example - but we don't have to change the Info.plist (meaning we don't have to re-authorize the app), and we don't have to relaunch the app.  Hooray!
* **Automatically grant Accessibility access when needed** - the recent change to launch the settings dialog got me thinking about ways to make the process of granting the Accessibility access even easier.  It turns out that, with administrator privileges, we can just write a record directly to the place where the system is looking for it, without asking the user to go in and do it on their own.  
  
I'm not a super big fan of the specific way this works (ugh, AppleScript), and I'm sure it's only a matter of time before Apple changes things again and we have to find another way around, and I wish we could have the system re-evaluate the privileges instead of relaunching the app... but the experience is otherwise pretty slick - just type in the password and it's done.  No muss, no fuss.

@akitchen, I'd like your eyes on this one, please!

### CODE REVIEW

**Comment by akitchen** [comment]
These are really nice improvements and I like the refactoring

Thumbs up!

**Comment by akitchen** [comment]
Looking forward to testing this out - I hope this method still works in the latest macOS

File: keycastr/KCAppController.m [hunk 1]
```diff
@@ -397,46 +466,74 @@ -(void) restartPanel:(NSAlert*)alert closedWithCode:(NSModalResponse)returnCode
...
+        // show icon in menu bar
+        [self createStatusItem];
+    }
+    else {
+        // hide icon in menu bar
+        [self deleteStatusItem];
+    }
+}
+
+-(void) observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NSDictionary<NSKeyValueChangeKey,id> *)change context:(void *)context {
```
Comment (akitchen) [inline keycastr/KCAppController.m]: It appears that NSKeyValueChangeKey is only available in the latest Xcode / macOS SDK 10.12+

If we want to maintain compatibility with "latest tools version minus one" I suggest we omit this from the method signature.

**Comment by sdeken** [comment]
Remember that part 12 days ago where I said "it's only a matter of time before Apple changes things again and we have to find another way around"?  Funny story, Apple changed this in Sierra so that you can't just write to the database directly anymore, even with root access.  So we'll have to either roll back to the way we were doing it before (sigh) or find a way to play nice with tccd (which locks this file now).

**Comment by akitchen** [comment]
Weird. This worked for me on macOS sierra (although, in the debugger, so maybe Xcode has additional privileges?).  Will test standalone too.  

... and there's another nit, which is that after changing back to "menu bar only" KeyCastr doesn't emit keystrokes until it is brought forward again, i.e. by opening the preferences window or about window.  I have some ideas around that one, but thought I'd mention it while we're here.

**Pull request merged**